users/views.py

```diff
@@ -0,0 +1,110 @@
+from django.shortcuts import render, redirect, get_object_or_404
+from django.views import View
+from django.contrib.auth.forms import AuthenticationForm, UserCreationForm
+from django.contrib.auth import authenticate, login, logout
+from django.contrib import messages
+from django.http import HttpResponse
+
+from django.contrib.auth.models import User
+
+from .forms.login_form import LoginForm
+from .forms.signup_form import SignupForm
+from .forms.profile_form import ProfileForm
+from .assets.login_authenticate import login_authenticate
+
+
+# Custom Error pages
+def error_404_view(request, exception):
+    return render(request, '404page.html')
+
+def error_500_view(request):
+    return render(request, '500page.html')
+
+def server_exception(request):
```

Comment:
@ahmadx16 Custom exception means to trigger any exception which is raised on any known case but isn't supported by available exceptions in that model/form/view etc. You need to implement one of that kind. E.g. for invalid form data you can have a custom exception with your custom message.